code/__HELPERS/icons.dm

```diff
@@ -692,7 +692,18 @@ The _flatIcons list is a cache for generated icon files.
 	var/image/copy
 	// Add the atom's icon itself, without pixel_x/y offsets.
 	if(!noIcon)
-		copy = image(icon=curicon, icon_state=curstate, layer=A.layer, dir=curdir)
+		if(curdir == SOUTH)
+			copy = image(curicon, curstate)
+		else
+			var/icon/dirIcon
+			if(curdir % 2 == 0)

```

Comment:
Pretty sure the Cardinal directions are even and diagonal are odd numbered, unless byond changed this in the last year since I really looked at ss13 code. 6 corresponds to south if I recall correctly. Although the "in cardinals" suggestion made is probably better than this line was.